### DISCUSSION TITLE
BETA: Reliability improvements, TB115 support

### DISCUSSION THREAD

**Discussion by jikamens** [Extended-Thunder/send-later#573] [General]
This is a beta release which includes a number of reliability and robustness improvements over the previous release as well as a first pass at supporting Thunderbird 115. This release _only_ supports Thunderbird 115 and newer; it does not support Thunderbird 102.

Caveats and known issues regarding Thunderbird 115:

* It is not possible in Thunderbird 115 for add-ons to add custom columns to the message list, so this functionality has been removed. It will be restored for a later Thunderbird release when it is possible. In the meantime, you can see a list of scheduled messages by clicking on the Send Later button in the main Thunderbird window.

* While the "Send Later" value for a message _is_ displayed in the message header in the main Thunderbird window, the display is currently a bit sloppy (the label and value appear on separate lines instead of a single line). This will be fixed in a later release when it is possible.

Bug fixes in this release (eventually these fixes will also be released in a version that is compatible with Thunderbird 102):

* The header row showing the Send Later value for a message now works reliably; previously it was intermittently failing.
* Send Later now correctly processes all drafts even when you have more than 100 drafts in a folder.
* The functionality for allowing other extensions to get Send Later's preferences was broken but is now working again.
* A bunch of intermittent timing issues have been resolved, making the add-on start up and function with fewer unexplained failures.
* As noted above, the custom Send Later column in the message list has been removed since it doesn't work in Thunderbird 115. However, two bugs in the custom column have been fixed and those fixes will be included in the next Thunderbird 102 release. In particular:
   * The column is now properly enabled / disabled when it should be.
   * The column is displayed on Thunderbird startup if the user has the Drafts folder selected as their initial folder.

<hr /><em>This discussion was created from the release <a href='https://github.com/Extended-Thunder/send-later/releases/tag/v10.0.0'>BETA: Reliability improvements, TB115 support</a>.</em>

**Comment by arbpen** [comment]
Thank you so much!  It seems the beta is working fine and just sent a slew of messages that were due to go out today.  There are more to go out 8/18/23, 8/19/23, 8/20/23, and 8/21/23.  I will check each day and if some are not performing as they should, I'll be back.

**Reply by RinTan12** [reply]
Thank you for your great work.  I have encountered the following issue:
After starting thunderbird, "send later" works correctly for the first time, but for the second and subsequent times, the email remains in the draft folder. FYI: even after the first time it works correctly, the number of waiting emails redly displayed on the icon outside the column is still 1 (not cleared)..

**Reply by jikamens** [reply]
@RinTan12 can you please reach out to me at send-later-support@extended-thunder.org so I can troubleshoot further with you? I don't want to bother other people in this thread with a back-and-forth. When you email please mention your GitHub username if it isn't obvious from your name or email address so I can connect your email to your comment here. Thanks!

**Reply by arbpen** [reply]
@RinTan12 I am experiencing the same thing.  I'm also writing to @jikamens
